src/Altinn.App.Core/Features/Action/SigningUserAction.cs

```diff
@@ -99,24 +101,28 @@ public async Task<UserActionResult> HandleAction(UserActionContext context)
         );
     }
 
-    private static bool ShouldSign(
+    private readonly record struct ShouldSignResult(
+        [property: MemberNotNullWhen(true, "SignatureDataType")] bool Value,
+        string? SignatureDataType
+    );
+
+    private static ShouldSignResult ShouldSign(
```

Comment:
This rewrite took a while to understand. Wouldn't a `private static string? GetDataTypeForSignature(---)` combined with `if (dataTypeForSignature is null)` do the same trick?

Reply:
Very good point, that ended up a lot simpler 😄 Latest commit implements that